File: Medium/198-house-robber/try-5.cpp
/*
 * Problem #198: House Robber
 * Difficulty: Medium
 * Submission: Try 5
 * status: Accepted
 * Language: cpp
 * Date: 22/03/2026, 16:38:50
 * Link: https://leetcode.com/problems/house-robber/
 */

class Solution {
public:
    int rob(vector<int>& nums) {
        int n = nums.size();
        vector<int> dp(n + 2);
        dp[n + 1] = 0;
        dp[n] = 0;
        for(int i = n - 1;i >= 0; i--){
            dp[i] = max(nums[i] + dp[i + 2], dp[i + 1]);
        }
        return dp[0];
    }
};

File: Medium/209-minimum-size-subarray-sum/try-5.py
/*
 * Problem #209: Minimum Size Subarray Sum
 * Difficulty: Medium
 * Submission: Try 5
 * status: Accepted
 * Language: python3
 * Date: 23/12/2024, 14:12:03
 * Link: https://leetcode.com/problems/minimum-size-subarray-sum/
 */

class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        s=0
        i=0
        mx=float('inf')
        for e in range(len(nums)):
            s+=nums[e]
            while(s>=target):
                mx = min( mx, e-i+1)
                s -= nums[i]
                i+=1
        
        return 0 if mx == float('inf') else mx
            

            
            




            


        
